tests/ludwig/config_validation/test_validate_config_misc.py

```diff
@@ -105,7 +100,7 @@ def test_config_features():
         }
 
         with pytest.raises(ConfigValidationError):
-            validate_config(config)
+            ModelConfig.from_dict(config)
```

Comment:
Something important I want to call out here: the `validate_config` (JSONSchema validation) and `ModelConfig.from_dict` serve different purposes in practice.

It's important that we test that `validate_config` works independent of any non-JSONSchema validation checks, because this tests that the JSONSchema validation itself is working correctly.

Ideally, the JSONSchema and `ModelConfig` checks behave identically (though the ModelConfig checks should be faster since it doesn't rely on compiling the JSONSchema, in theory). In practice, ModelConfig checks will be more extensive, and that's okay. But we shouldn't be losing coverage of the JSONSchema checks, because they are useful in cases where the JSONSchema is all we have.

Does that make sense?

So what I would like to see is either keeping these `validate_config` checks in place to verify correctness in the JSONSchema, or we add a new set of tests that check parity between JSONSchema and ModelConfig checks.

Reply:
That makes sense.

It sounds like we want to have separate tests, 1) specifically for the behavior of `check_schema` (that, notably, can be used in an environment without marshmallow) and 2) for any additional behavior that `ModelConfig.from_dict` would check.

We aspire to have 1) and 2) to have perfect parity, which is true for the most part with the notable exception of checks for interdependent parameters.